homeassistant/components/lyric/climate.py

```diff
@@ -356,56 +366,69 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set hvac mode."""
         _LOGGER.debug("HVAC mode: %s", hvac_mode)
         try:
-            if LYRIC_HVAC_MODES[hvac_mode] == LYRIC_HVAC_MODE_HEAT_COOL:
-                # If the system is off, turn it to Heat first then to Auto,
-                # otherwise it turns to.
-                # Auto briefly and then reverts to Off (perhaps related to
-                # heatCoolMode). This is the behavior that happens with the
-                # native app as well, so likely a bug in the api itself
-                if HVAC_MODES[self.device.changeableValues.mode] == HVACMode.OFF:
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[LYRIC_HVAC_MODE_COOL],
-                    )
-                    await self._update_thermostat(
-                        self.location,
-                        self.device,
-                        mode=HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                        autoChangeoverActive=False,
-                    )
-                    # Sleep 3 seconds before proceeding
-                    await asyncio.sleep(3)
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                    )
-                    await self._update_thermostat(
-                        self.location,
-                        self.device,
-                        mode=HVAC_MODES[LYRIC_HVAC_MODE_HEAT],
-                        autoChangeoverActive=True,
-                    )
-                else:
-                    _LOGGER.debug(
-                        "HVAC mode passed to lyric: %s",
-                        HVAC_MODES[self.device.changeableValues.mode],
-                    )
-                    await self._update_thermostat(
-                        self.location, self.device, autoChangeoverActive=True
-                    )
-            else:
+            match self._attr_thermostat_type:
+                case LyricThermostatType.TCC:
+                    await self._async_set_hvac_mode_tcc(hvac_mode)
+                case LyricThermostatType.LCC:
+                    await self._async_set_hvac_mode_lcc(hvac_mode)
+        except LYRIC_EXCEPTIONS as exception:
+            _LOGGER.error(exception)
```

Comment:
Nowadays we want to raise exceptions in service calls, either ServiceValidationError on bad user input or HomeAssistantError for other problems.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Reply:
That sounds reasonable to me, although this issue is pervasive throughout this file. Changing it here and nowhere else would make the behavior inconsistent, but changing everywhere else would introduce quite a bit of scope creep.

Would you be opposed if I opened a separate PR for that change?

Reply:
Created a separate PR for this: https://github.com/home-assistant/core/pull/105460

Let me know what you think. I'll have to do some conflict resolution depending on whether I merge that one or this one first.

Reply:
Yes, doing it separately is ok.